src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/FunctionCallContent.cs

```diff
@@ -45,6 +45,13 @@ public FunctionCallContent(string callId, string name, IDictionary<string, objec
     /// </summary>
     public IDictionary<string, object?>? Arguments { get; set; }
 
+    /// <summary>Gets or sets an additional context object specific to the provider that created the content.</summary>
+    /// <remarks>
+    /// If not <see langword="null"/>, the value of this property should be stored into any <see cref="FunctionResultContent"/>
+    /// created to represent the result of this function call.
+    /// </remarks>
+    public object? Context { get; set; }
```

Comment:
This is a bit weird because you might ask why not define `CallId` as type `object`, and then the provider can bundle any amount of ID/state in there that it likes, and the FRC creator would not need to write any extra code since they already know they have to provide the same ID on the result.

But then the call ID would typically get lost when round-tripping through serialization, and perhaps that's a problem we need to avoid, and is solved by having `CallId` be `string`. So my remaining question is: is it OK for `Context` to be lost when round-tripping through serialization? And if so, how can we communicate that to someone who's dealing with chat history serialization?